### PR TITLE
Daily build and smoke test improvements

### PR DESCRIPTION
**Pull request by isra-fel**
## Description

This PR contains 2 improvements to daily build and smoke test:
- Daily build was set to trigger every night 0AM, but it's UTC+0 time, so 8AM. I changed it to 18PM UTC+0, which is 2AM UTC+8
- Az.Maintenance usually fail in daily smoke test because of timeout. I let it retry automatically
- (Edit) Fixed an issue where `Get-AzPolicyEvent` may throw an error
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Review by BethanyZhou — APPROVED** [review]

**Comment by BethanyZhou** [comment]
@isra-fel , is this the reason why daily build didn't run as scheduled?
image
https://docs.microsoft.com/en-us/azure/devops/pipelines/process/scheduled-triggers?view=azure-devops&tabs=yaml#running-even-when-there-are-no-code-changes

File: .azure-pipelines/daily-build.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Please don't use ADO UI defined scheduled triggers because it takes precedence over YAML scheduled triggers.
 # https://docs.microsoft.com/en-us/azure/devops/pipelines/process/scheduled-triggers
 schedules:
-- cron: "0 0 * * *"
+- cron: "0 18 * * *" # 18PM UTC+0 aka 2AM UTC+8, every day
   displayName: Daily Midnight Build
```
Comment (BethanyZhou) [inline .azure-pipelines/daily-build.yml:5]:
```suggestion
- cron: "0 18 * * *" 
  displayName:  2:00 AM (UTC + 8:00) China Daily Build
```

**Comment by isra-fel** [comment]
@BethanyZhou  Hi Beisi I added a new commit to fix today's smoke test issue. Please review again. Thanks!

**Review by BethanyZhou — APPROVED** [review]
LGTM

**Pull request merged**